test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,213 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/search/fm_index/all.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+#include <seqan3/range/views/persist.hpp>
+#include <seqan3/range/views/rank_to.hpp>
+#include <seqan3/range/views/to.hpp>
+#include <seqan3/range/views/to_rank.hpp>
+
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/index.h>
+#endif
+
+static void arguments(benchmark::internal::Benchmark* b)
+{
+    for (int32_t length : {50, 5000, 50'000})
+    {
+        for (int32_t collection_size : {5})
+        {
+            b->Args({length, collection_size});
+        }
+    }
+}
+
+struct fm_index_seqan2;
+struct fm_index_seqan3;
+struct bi_fm_index_seqan2;
+struct bi_fm_index_seqan3;
+
+template <typename index_tag_t, typename rng_t>
+class sequence_generator
+{
+private:
+    size_t length{50'000};
+    size_t collection_size{5};
+
+    using alphabet_t = seqan3::innermost_value_type_t<rng_t>;
+    using inner_rng_t = std::conditional_t<seqan3::dimension_v<rng_t> == 1,
+                                             rng_t,
+                                             std::ranges::range_value_t<rng_t>>;
+    using fm_index_seqan3_t = decltype(seqan3::fm_index{std::declval<rng_t>()});
+    using bi_fm_index_seqan3_t = decltype(seqan3::bi_fm_index{std::declval<rng_t>()});
+
+    inner_rng_t make_single_seqan3(size_t seed = 0)
+    {
+        // The characters with rank 254 and 255 are reserved for the indices.
+        if constexpr (seqan3::alphabet_size<alphabet_t> > 254)
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed)
+                   | seqan3::views::persist
+                   | seqan3::views::to_rank
+                   | std::views::transform([] (uint8_t const r)
+                   {
+                       return (r > 253 ? 253 : r);
```

Comment:
more comprehensive is:
```suggestion
                       return std::max{r, 253u};
```

Reply:
`std::min` :) but yes, way more expressive 👍

Reply:
true! 😆